select/index.js

```diff
@@ -0,0 +1,313 @@
+                }
+                .irnmn-select__item {
+                    padding: 15px;
+                    border-bottom: 1px solid #e0e0e0;
+                    cursor: pointer;
+                }
+                .irnmn-select__item:hover,
+                .irnmn-select__item:focus {
+                    background-color: #D3F5FF;
+                }
+                .irnmn-select__item:last-child {
+                    border-bottom: none;
+                }
+                .irnmn-select__item--selected {
+                    background-color: #e6f3ff;
+                }
+                .irnmn-select__item--unselectable {
+                    color: #999;
+                    pointer-events: none;
+                    cursor: default;
+                }
+                .irnmn-select__item[aria-selected="true"] {
+                    font-weight: bold;
+                }
+                .irnmn-select__list {
+                    position: absolute;
+                    display: none;
+                    width: 100%;
+                    padding: 0;
+                    margin: 0;
+                    list-style-type: none;
+                    background-color: #fff;
+                    overflow: hidden;
+                    z-index: 1;
+                }
+                .irnmn-select__list--open {
+                    display: block;
+                    max-height: 400px;
+                    overflow: auto;
+                    top: 100%;
+                    bottom: auto;
+                }
+                .irnmn-select__list--open.open-upwards {
+                    top: auto;
+                    bottom: 100%;
+                }
+                .irnmn-select__item {
+                    border: 0;
+                    padding: 12px 24px;
+                }
+            </style>
+            <div class="irnmn-select ${this.selectedOption === null ? 'irnmn-select--unselected' : ''}" role="combobox" aria-expanded="${this.isOpen}" aria-haspopup="listbox" aria-labelledby="irnmn-select-header">
+                <div id="irnmn-select-header" class="irnmn-select__header" tabindex="0">
+                    ${this.selectedOption !== null ? this.options[this.selectedOption].name : this.headingText}
+                </div>
+                <ul id="option-list" class="irnmn-select__list ${this.isOpen ? 'irnmn-select__list--open' : ''}" role="listbox" aria-labelledby="irnmn-select-header">
+                    ${this.placeholder ? `
+                        <li class="irnmn-select__item irnmn-select__item--unselectable" role="option" aria-selected="false" tabindex="-1" disabled>
+                            ${this.placeholder}
+                        </li>` : ''}
+                    ${this.options
+                        .map(
+                            (option, index) => `
+                        <li class="irnmn-select__item" 
+                            role="option" 
+                            id="option-${index}" 
```

Comment:
This will create some redundancies if you use IDs. For instance, two booking components (bar and panel) will use the same Ids

Reply:
@samuelsilvapt using `data-index="${index}"` instead